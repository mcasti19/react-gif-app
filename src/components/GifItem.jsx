// SE debe agregar siempre el prop-types ya sea con yarn o npm y luego de eso se puede importar en el componente
import PropTypes from "prop-types";


export const GifItem = ( { title, url, id } ) => {

  return (
    <div className="card">
      <img src={ url } alt={ title } />
      <p>{ title }</p>
    </div>
  )
}


GifItem.propTypes = { // Leer la 1era linea
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}