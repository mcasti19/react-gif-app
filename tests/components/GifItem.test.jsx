// SE debe agregar siempre el prop-types ya sea con yarn o npm y 
// luego de eso se puede importar en el componente

import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe( 'Pruebas en <GifItem/>', () => {

  const title = 'Kakaroto';
  const url = 'https://www.udemy.com/';

  test( 'Debe hacer match con el snapshot', () => {
    const { container } = render( <GifItem title={ title } url={ url } /> )
    expect( container ).toMatchSnapshot();
    // screen.debug();
  } );

  test( 'Debe mostrar la img con el URL y el ALT indicado', () => {

    render( <GifItem title={ title } url={ url } /> );
    // screen.debug();

    //NO ES UNA BUENA PRACTICA HACERLO ASI
    // const altGif = screen.getByRole( 'img' ).alt;
    // const urlGif = screen.getByRole( 'img' ).src;


    // SIEMPRE QUE SE PUEDA HACER DESESTRUCTURACION ES LO MEJOR
    const { src, alt } = screen.getByRole( 'img' );

    expect( src ).toBe( url );
    expect( alt ).toBe( title );
  } );


  test( 'Debe mostrar el titulo en el componente', () => {

    render( <GifItem title={ title } url={ url } /> );
    expect( screen.getByText( title ) ).toBeTruthy();

  } );


  //ESTO FUE UNA PRUEBA MIA PERO NO FUNCIONA PORQUE NO HAY IDs O MAS PROPIEDADES PARA EVLUAR LA CLASE
  // test( 'Debe tener la clase CARD en el div principal', () => {
  //   render( <GifItem title={ title } url={ url } /> );
  //   screen.debug();
  //   const gifCard = screen.getByRole( 'div' );
  //   expect( gifCard ).toHaveClass( 'CARD' );

  // } );


} );