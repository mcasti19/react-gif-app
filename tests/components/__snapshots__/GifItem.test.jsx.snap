// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Kakaroto"
      src="https://www.udemy.com/"
    />
    <p>
      Kakaroto
    </p>
  </div>
</div>
`;
